File: DSA/SparshJain/Day_2_Q1.cpp
#include<iostream>
using namespace std;
int Check(int arr[],int time){
    int count=0;
    for(int i=0;i<time;i++){
        if(arr[i]<=0){
            count++;
        }
        
    }
    return count;
}
int main(){
    int thresh;
    cout<<"Input the threshold no of students";
    cin>>thresh;
    int time;
    cout<<"Enter the total no of arrival times \n";
    cin>>time;
    int At[time];
    cout<<"Input the array of arrival times\n";
    for(int i=0;i<time;i++){
        cin>>At[i];
        

    }
    int result=Check(At,time);
    if(result<thresh){
        cout<<"Yes";
    }
    else{
        cout<<"No";
    }

}

File: DSA/Ishita-Nigam/Day2/Day2_Q1.java
import java.util.Scanner;

public class Day2_Q1 {
    public static String isClassCancelled(int threshold, int[] arrivalTimes) {
        int onTimeCount = 0;
        for (int time : arrivalTimes) {
            if (time <= 0) {
                onTimeCount++;
            }
        }
        return onTimeCount < threshold ? "YES" : "NO";
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        System.out.println("Enter the threshold number of students:");
        int threshold = in.nextInt();

        System.out.println("Enter the arrival times of students (space-separated):");
        String[] input = in.next().split(" ");
        int[] arrivalTimes = new int[input.length];

        for (int i = 0; i < input.length; i++) {
            arrivalTimes[i] = Integer.parseInt(input[i]);
        }

        String result = isClassCancelled(threshold, arrivalTimes);
        System.out.println("Output: " + result);

        in.close();
    }
}
